tests/test_zonemap.py

```diff
@@ -45,74 +46,123 @@ def test_nonexistingzones():
 
 
 def test_errors(tmpdir, caplog):
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+    """Test in lyr parse function return correct errors"""
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
 foo
 """,
         encoding="utf-8",
     )
-    assert ecl2df.common.parse_zonemapfile(zonefile) is None
-    assert "Could not parse zonemapfile" in caplog.text
+    assert ecl2df.common.parse_lyrfile(lyrfile) is None
+    assert "Could not parse lyr file" in caplog.text
     assert "Failed on content: foo" in caplog.text
 
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
 valid 1-2
 foo 1 2 3
 """,
         encoding="utf-8",
     )
-    assert ecl2df.common.parse_zonemapfile(zonefile) is None
+    assert ecl2df.common.parse_lyrfile(lyrfile) is None
     assert "Failed on content: foo 1 2 3" in caplog.text
 
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
         """
-valid 1-2 stray
+foo 2-1
 """,
         encoding="utf-8",
     )
-    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(zonefile)) is None
-    assert "Failed on content: valid 1-2 stray" in caplog.text
+    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(lyrfile)) is None
+    assert "From_layer higher than to_layer" in caplog.text
+
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
+        """
+valid 1-2 #FFE5F7
+foo   3- 4 #FFGGHH
+""",
+        encoding="utf-8",
+    )
+    assert ecl2df.EclFiles(DATAFILE).get_zonemap(str(lyrfile)) is None
+    assert "Failed on content: foo   3- 4 #FFGGHH" in caplog.text
 
 
-def test_spaces_dashes(tmpdir):
-    """Ensure we support dashes around dashes"""
-    zonefile = tmpdir / "zonemap.lyr"
-    zonefile.write_text(
+def test_lyrlist_format(tmpdir):
+    """Ensure the lyr file is parsed correctly"""
+    lyrfile = tmpdir / "formations.lyr"
+    lyrfile.write_text(
+        """
+-- Some text
+'ZoneA'          1 -     5  #FFE5F7
+'ZoneB'          6-     10  --no color
+'ZoneC'          11-15    blue
+'ZoneD'         3          #fbb
+'ZoneE'         19     -20
+'ZoneF'         21-22  CORNFLOWERBLUE
+""",
+        encoding="utf-8",
+    )
+    lyrlist = ecl2df.common.parse_lyrfile(lyrfile)
+    # 0 - ZoneA
+    assert lyrlist[0]["from_layer"] == 1
```

Comment:
Can all these asserts be condensed into a more explicit test, like 
```
assert lyrlist == [{"from_layer": 1, "name": ZoneA, "color": "#FFE5F7}, {....}]
```

Reply:
Yes, I was wondering if that was possible. I will do it.